src/libs/ReportUtils.ts

```diff
@@ -1052,13 +1032,12 @@ function isSystemChat(report: OnyxEntry<Report>): boolean {
  * Only returns true if this is our main 1:1 DM report with Concierge.
  */
 function isConciergeChatReport(report: OnyxInputOrEntry<Report>): boolean {
-    const participantAccountIDs = Object.keys(report?.participants ?? {})
-        .map(Number)
-        .filter((accountID) => accountID !== currentUserAccountID);
-    return participantAccountIDs.length === 1 && participantAccountIDs[0] === CONST.ACCOUNT_ID.CONCIERGE && !isChatThread(report);
+    const participantAccountIDs = Object.keys(report?.participants ?? {});
+    return participantAccountIDs.length === 1 && Number(participantAccountIDs[0]) === CONST.ACCOUNT_ID.CONCIERGE && !isChatThread(report);
```

Comment:
Hi 👋 @hurali97 @hungvu193  , regarding this issue https://github.com/Expensify/App/issues/45036, in this PR we have removed the `accountID !== currentUserAccountID` filter for participants, which caused the regression where isConciergeChatReport returns a wrong value. A contributor suggested a fix that restores the old code. Before moving forward with the PR, can you please explain why we removed the filter and if adding back the old code will cause any issues?"

Reply:
Hey @jayeshmangwani 👋 

Oh snap, I removed this filter option as I thought it wasn't adding any value because later we are checking for `conciergeID` in return.

I tested the linked issue and I see that with these changes `isConciergeReport` return `false`, so it's better to bring back the filter only, we can ditch the map. Below is how the change will look like:

```ts

const participantAccountIDs = Object.keys(report?.participants ?? {}).filter((accountID) => Number(accountID) !== currentUserAccountID);

```

With this, we get true from `isConciergeReport` BUT I still don't see any "Book A Call" option. This also happened on Prod where I can't see this option.

Prod:


https://github.com/Expensify/App/assets/47336142/f7dd6a9f-91e1-4240-9f3d-a3d69e5dc237

<hr/>

I spent some time and this happens because `guideCalendarLink` is always `undefined` in ProfilePage for Concierge chat. So the `filter` change is not relevant to this bug BUT we should make this change either way.

Code to show the option is below, where we need both `isConcierge` and `guideCalendarLink` to be true, but even after the filter change, we get `guideCalendarLink` as undefined:
```tsx
        {isConcierge && guideCalendarLink && (
              <MenuItem
                     title={translate('videoChatButtonAndMenu.tooltip')}
                      icon={Expensicons.Phone}
                      isAnonymousAction={false}
                      onPress={SessionActions.checkIfActionIsAllowed(() => {
                       LinkActions.openExternalLink(guideCalendarLink);
                       })}
                    />
           )}
```

![Screenshot 2024-07-10 at 12 10 45 PM](https://github.com/Expensify/App/assets/47336142/2ff726bf-4a5a-4eb0-847d-440e6f67adbe)

Reply:
Thanks for the insight, and yea even though we revert the isConciergeChatReport logic we still cannot see the Book a call option because of the guideCalendarLink undefined value

Reply:
> I spent some time and this happens because guideCalendarLink is always undefined in ProfilePage for Concierge chat

@hurali97 The guideCalendarLink value may be undefined because we might be using an email that contains a '+'. With a normal email, we are able to get the guide link. 
Context: https://expensify.slack.com/archives/C01GTK53T8Q/p1720464122617119?thread_ts=1720457309.992509&cid=C01GTK53T8Q